From e9863fbb7f68e6c60432a5fedac4abcbe5658058 Mon Sep 17 00:00:00 2001
From: Marius Vollmer <mvollmer@redhat.com>
Date: Thu, 31 Oct 2013 15:55:07 +0200
Subject: [PATCH 510/512] logind: use correct "who" enum values with KillUnit.

NB This is already fixed differently in systemd 209 with the port to kdbus
---
 src/login/logind-dbus.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/login/logind-dbus.c b/src/login/logind-dbus.c
index c0b5e8f..6f3e442 100644
--- a/src/login/logind-dbus.c
+++ b/src/login/logind-dbus.c
@@ -2800,7 +2800,7 @@ int manager_kill_unit(Manager *manager, const char *unit, KillWho who, int signo
         assert(manager);
         assert(unit);
 
-        w = who == KILL_LEADER ? "process" : "cgroup";
+        w = who == KILL_LEADER ? "control" : "all";
         assert_cc(sizeof(signo) == sizeof(int32_t));
 
         r = bus_method_call_with_reply(
@@ -2816,7 +2816,7 @@ int manager_kill_unit(Manager *manager, const char *unit, KillWho who, int signo
                         DBUS_TYPE_INT32, &signo,
                         DBUS_TYPE_INVALID);
         if (r < 0) {
-                log_error("Failed to stop unit %s: %s", unit, bus_error(error, r));
+                log_error("Failed to kill unit %s: %s", unit, bus_error(error, r));
                 return r;
         }
 
-- 
1.8.4.5

